skaha/src/main/java/org/opencadc/skaha/session/DeleteAction.java

```diff
@@ -105,24 +105,33 @@ public void doAction() throws Exception {
                         "USERID:.metadata.labels.canfar-net-userid"};
                         
                 String session = execute(getSessionCMD);
-                if (!StringUtil.hasText(session)) {
-                    throw new ResourceNotFoundException(sessionID);
-                }
-                String[] lines = session.split("\n");
-                if (lines.length != 1) {
-                    throw new IllegalStateException("found multiple sessions with id " + sessionID);
+                if (StringUtil.hasText(session)) {
+                    String[] lines = session.split("\n");
+                    if (lines.length > 0) {
+                        // sessionID was added to desktop-app. This resulted in the 
+                        // above kubectl command returning desktop-app as well. We 
+                        // want to ignore them as we pick the session to be deleted.
+                        for (String line : lines) {
+                            String[] parts = line.split("\\s+");
+                            if (!TYPE_DESKTOP_APP.equals(parts[0])) {
+                                String sessionUserId = parts[1];
+                                if (!userID.equals(sessionUserId)) {
+                                    throw new AccessControlException("forbidden");
+                                }   
+    
+                                String type = parts[0];
```

Comment:
The type could be extracted (this line) before it's used (parts[0]) on line 116.

Reply:
Updated.